packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/500.tsx

```diff
@@ -0,0 +1,57 @@
+import Head from 'next/head';
+import { GraphQLErrorPagesService, SitecoreContext } from '@sitecore-jss/sitecore-jss-nextjs';
+import { SitecorePageProps } from 'lib/page-props';
+import Layout from 'src/Layout';
+import { componentFactory } from 'temp/componentFactory';
+import { GetStaticProps } from 'next';
+import config from 'temp/config';
+import { siteResolver } from 'lib/site-resolver';
+
+/**
+ * Rendered in case if we have 500 error
+ */
+const ServerError = (): JSX.Element => (
+  <>
+    <Head>
+      <title>500: Server Error</title>
+    </Head>
+    <div style={{ padding: 10 }}>
+      <h1>500 Internal Server Error</h1>
+      <p>There is a problem with the resource you are looking for, and it cannot be displayed.</p>
+      <a href="/">Go to the Home page</a>
+    </div>
+  </>
+);
+
+const Custom500 = (props: SitecorePageProps): JSX.Element => {
+  if (!(props && props.layoutData)) {
+    return <ServerError />;
+  }
+
+  return (
+    <SitecoreContext componentFactory={componentFactory} layoutData={props.layoutData}>
+      <Layout layoutData={props.layoutData} headLinks={props.headLinks} />
+    </SitecoreContext>
+  );
+};
+
+export const getStaticProps: GetStaticProps = async (context) => {
+  const site = siteResolver.getByName(config.jssAppName);
+  const errorPagesService = new GraphQLErrorPagesService({
+    endpoint: config.graphQLEndpoint,
+    apiKey: config.sitecoreApiKey,
+    siteName: site.name,
+    language: context.locale || config.defaultLanguage,
```

Comment:
Should we use _defaultLocale_ provided in a context? Since we setup default language in next.config

```suggestion
    language: context.locale || context.defaultLocale,
```